### PR TITLE
sql: implement mz_internal.mz_get_subsources

### PR DESCRIPTION
**Pull request by sploiselle**
Implement an internal builtin function to return an objects subsources and their types.

### Motivation

This PR adds a known-desirable feature. We want this to improve the UX of customers performing introspection for PG subsources + finding their progress subsources.

### Checklist

- [x] This PR has adequate test coverage / QA involvement has been duly considered.
- [x] This PR has an associated up-to-date [design doc](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/README.md), is a design doc ([template](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/design/00000000_template.md)), or is sufficiently small to not require a design.
  
- [ ] If this PR evolves [an existing `$T ⇔ Proto$T` mapping](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/command-and-response-binary-encoding.md) (possibly in a backwards-incompatible way), then it is tagged with a `T-proto` label.
- [ ] If this PR will require changes to cloud orchestration, there is a companion cloud PR to account for those changes that is tagged with the release-blocker label ([example](https://github.com/MaterializeInc/cloud/pull/5021)).
  
- [ ] This PR includes the following [user-facing behavior changes](https://github.com/MaterializeInc/materialize/blob/main/doc/developer/guide-changes.md#what-changes-require-a-release-note):
  -

### CODE REVIEW

File: src/sql/src/func.rs [hunk 1]
```diff
@@ -3148,6 +3148,25 @@ pub static MZ_INTERNAL_BUILTINS: Lazy<BTreeMap<&'static str, Func>> = Lazy::new(
             // message is the second argument.
             params!(Any, String) => VariadicFunc::ErrorIfNull => Any, oid::FUNC_MZ_ERROR_IF_NULL_OID;
         },
+        "mz_get_subsources" => Table {
+            params!(String) => sql_impl_table_func("
+                SELECT
+                    mz_internal.mz_global_id_to_name(d.object_id) AS source,
```
Comment (jkosh44) [inline src/sql/src/func.rs:3154]: Can you add a test that includes this column and is non-empty?

File: test/sqllogictest/mz-global-id-funcs.slt [hunk 2]
```diff
@@ -142,3 +142,8 @@ query T
 SELECT mz_internal.mz_name_to_global_id(null, ARRAY['pg_catalog'], 'int4')
 ----
 s20
+
+# Tables have no subsources
+query TTT
+SELECT * FROM mz_internal.mz_get_subsources('t');
+----
```
Comment (jkosh44) [inline test/sqllogictest/mz-global-id-funcs.slt:149]: Can you add a test where the table has some views or indexes to make sure those are mistakenly being show.

**Review by jkosh44 — APPROVED** [review]
LGTM
